app/_src/gateway/reference/wasm.md

```diff
@@ -0,0 +1,141 @@
+deployed in text format. Filters must be compiled from source code
+into a single binary file with the `.wasm` extension.
+
+In order to be usable in {{site.base_gateway}}, all filter files must be present
+on disk in a single directory, denoted by the `wasm_filters_path` configuration
+directive.
+
+### Filter chain
+
+A filter chain is the database entity representing one or more filters executed
+for each request to a particular service or route, each one with its
+configuration.
+
+Example:
+
+```yaml
+---
+filter-chains:
+  name: my-filter-chain
+  # filter chains can be toggled on/off for testing
+  enabled: true
+  filters:
+    # this maps to $wasm_filters_path/my_filter.wasm
+    - name: my_filter
+      # at runtime, configuration is passed to the filter as a byte array
+      #
+      # filter configuration is inherently untyped--schema and
+      # encoding/serialization are left up to the filter code itself
+      config: "foo=bar baz=bat"
+
+    - name: my_other_filter
+      # individual filters within a chain can be toggled on/off for testing
+      enabled: false
+      config: >-
+          {
+            "some_settings": {
+              "a": "b",
+              "c": "d"
+            },
+            "foo": "bar"
+          }
+
+    # the same filter may occur more than once in a single chain, and all   entries will be
+    # executed
+    - name: my_filter
+      config: "foo=123 baz=456"
+```
+
+
+#### Relationships
+
+Filter chains **must** be linked to a service or route, and as of this writing,
+the relationship is one-to-one:
+
+* a service or route may only be linked to a single filter chain
+* a filter chain may only be linked to a single service or route
+
+## Filter execution behavior
+
+The filter execution plan for a given request is determined during the `access`
+phase, after Lua plugins execution. If the Service entity for the request
+is linked to a filter chain, the filters are executed. If the Route entity that
+was matched to the request is linked to a filter chain, its filters are _also_
+executed, concatenating the list of filters from the matched Service and Route filter chains to produce the final execution plan. You can inspect the full filter execution plan for a given route via the Admin API, through the endpoint `/routes/{route name or id}/filters/enabled`.
+
+Filters are always executed in the order in which they are defined in the filter
+chain. The same filter may appear in the execution plan multiple times, either coming from the Service and Route chains, or even within the same filter chain. Each entry will execute with its own configuration.
+
+### Can Lua plugins and Proxy-Wasm filters be used at the same time?
+
+Yes -- Keep in mind though that Lua plugins are executed earlier in the request
+lifecycle, **before** filter chain linkage is assessed. So, if a Lua plugin
+terminates the request early during the `access` phase, no Proxy-Wasm filters will
+be executed.
```

Comment:
I think it should be further expanded: it isn't so much that Lua plugins are executed "earlier" in the request, it is more so that *each Nginx phase* is executed in 2 times:
1) Lua plugins for phase X
2) Wasm filters for phase X

Reply:
Okay, I amended this in 89c7e612abc3d07cb26ca3f1fbb38eab2ef8ba02. Please re-check it for accuracy, because at this point I think I have some blind spots with regards to how execution works after the access phase.